lib/seattleflu/id3c/cli/command/etl/redcap_det_uw_reopening.py

```diff
@@ -412,16 +418,29 @@ def create_encounter_id(record: REDCapRecord, is_followup_encounter: bool) -> st
         f'{redcap_repeat_instance}{encounter_identifier_suffix}'
 
 
-def get_collection_date(record: dict, collection_method: CollectionMethod) -> Optional[str]:
+def get_collection_date(record: REDCapRecord, collection_method: CollectionMethod) -> Optional[str]:
     """
     Determine sample/specimen collection date from the given REDCap *record*.
     """
+    # For all surveys, try the survey _timestamp field (which is in Pacific time)
+    # before custom fields because the custom fields aren't always populated and when
+    # they are populated they use the browser's time zone.
+    collection_date = None
+
     if collection_method == CollectionMethod.KIOSK:
-        return record["nasal_swab_q"]
+        collection_date = extract_date_from_survey_timestamp(record, "kiosk_registration_4c7f") or record.get("nasal_swab_q")
+
     elif collection_method == CollectionMethod.SWAB_AND_SEND:
-        return record["date_on_tube"] or record["kit_reg_date"] or record["back_end_scan_date"]
-    else:
-        return None
+        # Don't use `date_on_tube` if it's in the future
+        if record.get("date_on_tube") and datetime.strptime(record.get("date_on_tube"), "%Y-%m-%d").date() <= datetime.today().date():
+            collection_date = record.get("date_on_tube")
```

Comment:
To detect bad/future dates in the date_on_tube, I'm not sure "today" is the right comparison point. That is dependent on when the ETL runs, and could give different data accordingly. It seems more accurate to compare the date_on_tube with dates from an instruments like kit reg.

Also, I know this issue was raised for HCT specifically, but I think its important to avoid drift in this type of logic across the various project ETLs (scan and childcare come to mind). I'd suggest putting this specific change to ignore future dates as a single commit across the etls, split off from the present commit here to pull dates from survey timestamps.

Reply:
OK, I've removed the handling of future date_on_tube values from this PR. I've updated the Trello card to apply to all relevant ETLs. We can make that change across the ETLs at a future point.

Reply:
Ok. Also, sorry if this wasn't clear - I was suggesting a different commit not a different PR, but if you want to handle that completely separately that's ok too. Just remember to update the commit message for 7f5f332569ad54c0a3068ab1b1b20dd8d823da01 to remove the language about future dates and this should be good to go on my end.

Reply:
I don't have time right now to touch all the ETLs to add the future date check in.
I've updated the commit message.